### PR TITLE
Add custom validator tutoral

### PR DESCRIPTION
**Pull request by asmacdo**
Covers the `--alpha-select-external` usage, writing a custom validator
from scratch, and how to use validations in `operator-framework/api` at
arbitrary versions.

Signed-off-by: Austin Macdonald <austin@redhat.com>

### CODE REVIEW

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (jmrodri) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:74]: might be able to just do `go run main.go`

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (jmrodri) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:84]: +1

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (jmrodri) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:97]: Good explanation. I don't think I thought about even calling that out. Nice!

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:8]:
```suggestion
Operator authors can now use "external" validators with the
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:10]:
```suggestion
`--alpha-select-external` flag. This feature enables Operator authors,
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:19]:
```suggestion
$ operator-sdk bundle validate --alpha-select-external \
path/validator1:path/validator2
```
Not sure if you want to include the terminal prompt to specify that only user privelages are required. And I am not sure if you break long CLI commands for readability. If so, be sure the other CLI commands for consistency.

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:71]:
```suggestion
its own, this validator prints a JSON representation of `ManifestResult`.
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:70]: Would it make sense to move the sentence, "These custom validators can be written in any language" up to the _Writing Custom Validators_ section that introduces the topic? It feels like you are burying the lede here and this should be emphasized.

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:83]:
```suggestion
As long as the STDOUT follows the `ManifestResult` spec, it can be used
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:117]:
```suggestion
You can also validate using a different version of the `operator-framework/api` 
than what is built into the `operator-sdk` binary. You can 
import the `validation` package from the `api` repository
at an arbitrary version. 
```
Typo correction and a suggestion to break up the long sentence.

**Review by michaelryanpeter — COMMENTED** [review]
I left a few suggestions for style and consistency. Please feel free to take what is useful and disregard the rest. Great job!

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [reply website/content/en/docs/advanced-topics/custom-bundle-validation.md:117]: Would it help the user to have direct links to the repos and packages in this sentence?

**Comment by jmrodri** [comment]
@asmacdo any more progress with this?

**Comment by jmrodri** [comment]
@asmacdo where does this PR stand?

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (everettraven) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:21]: Should we move the flag down with the flag arguments for better readability?

Something like:
```suggestion
$ operator-sdk bundle validate \ 
--alpha-select-external path/validator1:path/validator2
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (everettraven) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:266]: Remove this
```suggestion
	if err != nil {
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (michaelryanpeter) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:129]: Since the code that follows is more of an example config file, not something a user is editing or a command to run, would it make sense to add a title to the config? Something like: **Example `main.go` file**.

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (everettraven) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:86]: Should we reiterate this point here? It is already specified that this is a requirement here: https://github.com/operator-framework/operator-sdk/pull/5881/files#diff-921c84e16a264fb65ea7d53da616dbe82e6f180cfa75bf4e5f55244f61dfef7cR28

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (everettraven) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:212]: Typos:
```suggestion
// generally useful please file an issue to move this to a reusable library.
```

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (asmacdo) [reply website/content/en/docs/advanced-topics/custom-bundle-validation.md]: good point, removed

**Review by everettraven — APPROVED** [review]
This looks good to me, but holding off on an LGTM so that others can review

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (rashmigottipati) [inline website/content/en/docs/advanced-topics/custom-bundle-validation.md:312]: Link seems to be broken.

Comment (rashmigottipati) [reply website/content/en/docs/advanced-topics/custom-bundle-validation.md:312]:
```suggestion
[errors-pkg]: https://github.com/operator-framework/api/tree/master/pkg/validation/errors
```

Comment (laxmikantbpandhare) [reply website/content/en/docs/advanced-topics/custom-bundle-validation.md:312]: Yes, not able to access the errors-pkg link. Rest Looks good to me.

**Review by laxmikantbpandhare — APPROVED** [review]
/lgtm after link broken fix.

File: website/content/en/docs/advanced-topics/custom-bundle-validation.md
Comment (asmacdo) [reply website/content/en/docs/advanced-topics/custom-bundle-validation.md:312]: I blindly used the pkg import from go rather than the actual human link :/ Fixed!

**Pull request merged**
